queue/group.go

```diff
@@ -158,3 +177,360 @@ func (g *localQueueGroup) Close(ctx context.Context) {
+		q, err := g.startProcessingRemoteQueue(ctx, coll)
+		if err != nil {
+			catcher.Add(errors.Wrap(err, "problem starting queue"))
+		} else {
+			g.queues[idFromCollection(g.prefix, coll)] = q
+			g.ttlMap[idFromCollection(g.prefix, coll)] = time.Now()
+		}
+	}
+	if catcher.HasErrors() {
+		return nil, catcher.Resolve()
+	}
+
+	if opts.PruneFrequency > 0 && opts.TTL > 0 {
+		go func() {
+			defer recovery.LogStackTraceAndContinue("panic in remote queue group ticker")
+			ticker := time.NewTicker(opts.PruneFrequency)
+			defer ticker.Stop()
+			for {
+				select {
+				case <-ctx.Done():
+					return
+				case <-ticker.C:
+					g.Prune(ctx)
+				}
+			}
+		}()
+	}
+	return g, nil
+}
+
+func (g *remoteQueueGroup) startProcessingRemoteQueue(ctx context.Context, coll string) (Remote, error) {
+	q, err := g.constructor(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "problem starting queue")
+	}
+	d, err := OpenNewMongoDriver(ctx, coll, g.mongooptions, g.client)
+	if err != nil {
+		return nil, errors.Wrap(err, "problem opening driver")
+	}
+	if err := q.SetDriver(d); err != nil {
+		return nil, errors.Wrap(err, "problem setting driver")
+	}
+	if err := q.Start(ctx); err != nil {
+		return nil, errors.Wrap(err, "problem starting queue")
+	}
+	return q, nil
+}
+
+func (opts RemoteQueueGroupOptions) validate() error {
+	if opts.Client == nil {
+		return errors.New("must pass a client")
+	}
+	if opts.Constructor == nil {
+		return errors.New("must pass a constructor")
+	}
+	if opts.TTL < 0 {
+		return errors.New("ttl must be greater than or equal to 0")
+	}
+	if opts.TTL > 0 && opts.TTL < time.Second {
+		return errors.New("ttl cannot be less than 1 second, unless it is 0")
+	}
+	if opts.PruneFrequency < 0 {
+		return errors.New("prune frequency must be greater than or equal to 0")
+	}
+	if opts.PruneFrequency > 0 && opts.TTL < time.Second {
+		return errors.New("prune frequency cannot be less than 1 second, unless it is 0")
+	}
+	if (opts.TTL == 0 && opts.PruneFrequency != 0) || (opts.TTL != 0 && opts.PruneFrequency == 0) {
+		return errors.New("ttl and prune frequency must both be 0 or both be not 0")
+	}
+	if opts.MongoOptions.DB == "" {
+		return errors.New("db must be set")
+	}
+	if opts.Prefix == "" {
+		return errors.New("prefix must be set")
+	}
+	if opts.MongoOptions.URI == "" {
+		return errors.New("uri must be set")
+	}
+	return nil
+}
+
+func getExistingCollections(ctx context.Context, client *mongo.Client, db, prefix string) ([]string, error) {
+	c, err := client.Database(db).ListCollections(ctx, bson.M{"name": bson.M{"$regex": fmt.Sprintf("^%s.*", prefix)}})
+	if err != nil {
+		return nil, errors.Wrap(err, "problem calling listCollections")
+	}
+	defer c.Close(ctx)
+	var collections []listCollectionsOutput
+	for c.Next(ctx) {
+		elem := &listCollectionsOutput{}
+		if err := c.Decode(elem); err != nil {
+			return nil, errors.Wrap(err, "problem parsing listCollections output")
+		}
+		collections = append(collections, *elem)
```

Comment:
can you remove the & from line 337, put it on line 338 and then remove the * here.